haystack/core/pipeline/async_pipeline.py

```diff
@@ -31,14 +31,13 @@ class AsyncPipeline(PipelineBase):
     """
 
     @staticmethod
-    async def _run_component_async(  # pylint: disable=too-many-positional-arguments
+    async def _run_component_async(
         component_name: str,
         component: Dict[str, Any],
         component_inputs: Dict[str, Any],
         component_visits: Dict[str, int],
-        max_runs_per_component: int = 100,
         parent_span: Optional[tracing.Span] = None,
-    ) -> Dict[str, Any]:
+    ) -> Mapping[str, Any]:
```

Comment:
I made this change to `Mapping` since that's what our Component protocol was updated to. And Dict is just a subset of Mapping so this won't affect anything, just provide better typing support.